tests/run-macros/f-interpolator-tests.scala

```diff
@@ -0,0 +1,195 @@
+      try { println(f"""Bob is ${f"$n"} years ${s"$old"}!""") } catch catcher
+      try { println(f"""Bob is ${f"$n%2d"} years ${s"$old"}!""") } catch catcher
+      try { println(f"""Bob is ${s"$n%2d"} years ${s"$old"}!""") } catch catcher
+    }
+
+    test1(1)
+    test1(12)
+    test1(123)
+
+    test2(10.0f)
+    test2(13.345f)
+
+    test3(1)
+    println("===============")
+    test3(12)
+    println("===============")
+    test3(123)
+  }
+
+  def generalArgsTests = {
+
+    def booleanTest(b : Boolean) = println(f"The boolean is $b%b %<b")
+
+    def hTest(arg : Int) = println(f"$arg%h")
+
+    def stringTest(s : String) = println(f"The string is $s%s $s%s %1$$s")
+
+    def noFormatHasStringDefault(s : String) = println(f"The string is $s")
+
+    booleanTest(false)
+    booleanTest(true)
+
+    hTest(10)
+
+    stringTest(null)
+    stringTest("")
+    stringTest("string1")
+
+    noFormatHasStringDefault(null)
+    noFormatHasStringDefault("")
+    noFormatHasStringDefault("string2")
+  }
+
+  def characterArgsTests = {
+
+    def charTest(c : Char) = println(f"The unicode character is $c%c")
+
+    charTest('c')
+  }
+
+  def integralArgsTests = {
+
+    def decimalIntegerTest(i : Int) = println(f"The decimal integer is $i%d, $i%2d, $i%d+1 = ${i+1}%d")
+
+    def octalIntegerTest(i : Int) = println(f"The octal integer is $i%o")
+
+    def hexadecimalIntegerTest(i : Int) = println(f"The hexadecimal integer is $i%x")
+
+    decimalIntegerTest(2)
+
+    octalIntegerTest(2)
+    octalIntegerTest(8)
+
+    hexadecimalIntegerTest(2)
+    hexadecimalIntegerTest(32)
+  }
+
+  def floatingPointArgsTests = {
+    def scientificNotationTest(f : Float) = println(f"The scientific notation is $f%e")
+
+    def decimalFloatingPointTest(f : Float) = println(f"The decimal floating point is $f%f, $f%3.2f, $f%15.3f")
+
+    def noPointFloatingPointTest(f : Float) = println(f"The decimal floating point is $f%2.0f")
+
+    def gTest(f : Float) = println(f"The float value is $f%g")
+
+    def aTest(f : Float) = println(f"The float value is $f%a")
+
+    scientificNotationTest(2f)
+    scientificNotationTest(0.543f)
+
+    decimalFloatingPointTest(1234.5678f)
+
+    noPointFloatingPointTest(10.0f)
+
+    val i = Math.pow(10.0, -4).floatValue
+
+    gTest(i)
+    gTest(i/10.0f)
+
+    aTest(-i/10.0f)
+    aTest(+0f)
+    aTest(Float.NaN)
+    aTest(Float.PositiveInfinity)
+    aTest(Float.NegativeInfinity)
+  }
+
+  def dateArgsTests = {
+    import java.text.SimpleDateFormat
+
+    val sdf = new SimpleDateFormat("dd-M-yyyy hh:mm:ss")
+    val dateInString = "31-08-1982 10:20:56"
+    val date = sdf.parse(dateInString)
+
+    println(f"$date%tH")
+    println(f"$date%tI")
+    println(f"$date%tk")
+    println(f"$date%tl")
+    println(f"$date%tM")
+    println(f"$date%tS")
+    println(f"$date%tL")
+    println(f"$date%tN")
+    println(f"$date%tp")
+    // println(f"$date%tz")
```

Comment:
Why are these commented out?

Reply:
I commented these lines out because the date format depends on the country, the time zone, etc. Therefore, it may create some problem, depending on where we run these tests.

Reply:
I have tried to set the local, hoping it will work.

Reply:
I see. Maybe add a comment.

Reply:
It does not work so I have only commented on the tests that are language dependent, i.e. the ones that should output "Tuesday" or "Mardi".